installer/pkg/config-generator/generator.go

```diff
@@ -61,6 +62,76 @@ func New(cluster config.Cluster) ConfigGenerator {
 	}
 }
 
+// maoOperatorConfig contains configuration for mao managed stack
+// TODO(enxebre): move up to "github.com/coreos/tectonic-config
+type maoOperatorConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	Provider        string         `json:"provider"`
+	AWS             *awsConfig     `json:"aws"`
+	Libvirt         *libvirtConfig `json:"libvirt"`
+}
+
+type libvirtConfig struct {
+	ClusterName string `json:"clusterName"`
+	URI         string `json:"uri"`
+	NetworkName string `json:"networkName"`
+	Replicas    int    `json:"replicas"`
+}
+
+type awsConfig struct {
+	ClusterName           string `json:"clusterName"`
+	ClusterID             string `json:"clusterID"`
+	Region                string `json:"region"`
+	AvailabilityZone      string `json:"availabilityZone"`
+	Image                 string `json:"image"`
+	ReleaseChannel        string `json:"releaseChannel"`
+	ContainerLinuxVersion string `json:"containerLinuxVersion"`
+	Replicas              int    `json:"replicas"`
+}
+
+func (c *ConfigGenerator) maoConfig() *maoOperatorConfig {
+	cfg := maoOperatorConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "machineAPIOperatorConfig",
+		},
+	}
+
+	clVersion := string(c.Version)
+	if clVersion == "latest" {
+		clVersion = "1800.7.0"
```

Comment:
This looks like a hack for testing, and is probably why we're [getting the CL AMI in the test output][1].  With work to drop CL so ckose (#180), and [no public URL for RHCOS AMIs (yet)][2], we probably need to use an AWS request [along these lines][3] (but for RHCOS and in Go).  I think the lookup utility should probably go into its own, new package (`pkg/rhcos/ami`?).

[1]: https://github.com/openshift/installer/pull/119/commits/82ea9cac9ddd3c8ba5b074bcaaff60051c367004#r215033988
[2]: https://github.com/openshift/release/pull/1344#issuecomment-418449544
[3]: https://github.com/openshift/installer/blob/f2e6b3ed56aa114cd263369fd626d6b0e493e6ed/modules/aws/ami/main.tf#L18

Reply:
@crawford @wking yeah this is just mimicking current behaviour, otherwise Jenkins tests would fail. As pointed by https://github.com/openshift/installer/blob/f2e6b3ed56aa114cd263369fd626d6b0e493e6ed/modules/aws/ami/main.tf#L18 masters are still retrieving the image based on "coreos*" filters and the aws actuator is doing the same based on the info passed here
https://github.com/openshift/machine-api-operator/pull/40/files#diff-de11afe02c707ee4595394c18e808430R31
https://github.com/openshift/cluster-api-provider-aws/blob/master/cloud/aws/actuators/machine/actuator.go#L212
We'll update this to satisfy the new image once we drop CL, otherwise is there anything you'd like us to do now?

Reply:
>  With the switch to RHCOS so close, I'm fine with the AMI-lookup package (pkg/rhcos/ami?) being a stub that just returns a hard-coded recent RHCOS AMI (this one?).

I've mocked this up in https://github.com/wking/openshift-installer/commit/89a220abdfd4ddc517dcd8f98814be69f0c4d98d.  If that looks useful, feel free to pull it into this PR (or I can file another PR to get it landed, and you can rebase this one on top, whichever is easier).

Reply:
Is that image public? It doesn't seem to be. If it's not, this won't be usable without an override. Is that acceptable?

Reply:
> Is that image public?

Nope (although the plan is to make both the images and the lookup URL public in "the near future" :p.

> If it's not, this won't be usable without an override. Is that acceptable?

I think so.  It should only be for the brief window before the RHCOS AMIs and lookup URL become public.  And it will work in our Prow AWS account (which has access to the RHCOS AMIs), so CI will work.  And it will work for the Go unit tests too (which is what you need for this PR).

Reply:
Okay, I can do that. So, currently the config references are to Container Linux, and there is validation on the channel name that would not allow "tested". Do you want me to just fix the validation to allow that value, or change the config format somehow? This honestly feels pretty far outside the scope of this PR.

Reply:
> So, currently the config references are to Container Linux, and there is validation on the channel name that would not allow "tested".

Ah, I was hoping we could leave the Container Linux stuff alone, replace it for your tests with an overridden RHCOS AMI, and eventually just kill the Container Linux stuff once we get #180 landed.  I'll take a closer look at this PR to see about wiring my RHCOS-lookup function into what you have here.

> This honestly feels pretty far outside the scope of this PR.

Dropping Container Linux is a near-term goal with several bits moving in parallel.  Sorry for the flux :p.

Reply:
Cool. No worries. Thanks. So, the lookup function is the easy bit. It's more about the config. We could just ignore the config in this bit of the generator for now and always use "tested". We're returning a constant string from the lookup anyway. Would you be okay with that?

Reply:
Pushed a change that does this. Testing it now. Let me know if this acceptable for the time being.

Reply:
> We could just ignore the config in this bit of the generator for now and always use "tested". We're returning a constant string from the lookup anyway. Would you be okay with that?

Yeah, that's fine with me.  Does this mean we can drop `clVersion`?

Reply:
Yeah, I meant to do that and just missed it. I'll drop it and see if I can fix the tests.